lms/services/canvas_api/client.py

```diff
@@ -241,23 +242,28 @@ def list_files(self, course_id, sort="position"):
             log.exception(
                 "Duplicates files found in Canvas courses/{course_id}/files endpoint"
             )
-
-        # Notify that we've found some files
-        self._request.registry.notify(
-            FilesDiscoveredEvent(
-                request=self._request,
-                values=[
-                    {
-                        "type": "canvas_file",
-                        "course_id": course_id,
-                        "lms_id": file["id"],
-                        "name": file["display_name"],
-                        "size": file["size"],
-                    }
-                    for file in files
-                ],
-            )
+        application_instance = self._request.find_service(
+            name="application_instance"
+        ).get_current()
+
+        bulk_upsert(
+            self._request.db,
+            File,
+            [
+                {
+                    "application_instance_id": application_instance.id,
+                    "type": "canvas_file",
+                    "course_id": course_id,
+                    "lms_id": file["id"],
+                    "name": file["display_name"],
+                    "size": file["size"],
+                }
+                for file in files
+            ],
+            ["application_instance_id", "lms_id", "type", "course_id"],
+            ["name", "size"],
```

Comment:
I wonder if we need a `FileService.upsert()` method that both `BlackboardAPIClient` and `CanvasAPIClient` can call? Might remove some duplication here